### ISSUE TITLE
NSCP Crashs on (WEBGUI-)Modules request

### ISSUE THREAD

**Issue by itsul** [mickem/nscp#322]
Hi Mickem,

while clicking on "Modules" in the WebGUI the NSCP stop working.

Version: (0.5.0.60 2016-09-05)

Output in test mode on console:
D       core Found cached CheckDisk
D       core Found cached CheckEventLog
D       core Found cached CheckExternalScripts
D       core Found cached CheckHelpers
D       core Loading C:/Program Files\NSClient++\modules\CheckLogFile.dll
D       core Loading C:/Program Files\NSClient++\modules\CheckNet.dll
D       core Found cached CheckNSCP
D       core Found cached CheckSystem
INF: Detected crash...
INF: Running: "C:\Program Files\NSClient++\reporter.exe" "archive" "C:\Users\user\AppData\Local\Temp\eb2b81e6-58d7-4120-bab4-c49e8e03d952.dmp" "NSClient++" "0.5.0.60" "2016-09-05" "C:\Program Files\NSClient++/crash-dumps"
D       core Loading C:/Program Files\NSClient++\modules\CheckTaskSched.dll
D       core Loading C:/Program Files\NSClient++\modules\CheckWMI.dll
D       core Found cached CommandClient
D       core Loading C:/Program Files\NSClient++\modules\DotnetPlugins.dll
ERR: Service not started, not restarting...

Crash Log / DUMP:
[b6a6a05f-c50f-4fba-9fb3-e3a42bce4afe.dmp.txt](https://github.com/mickem/nscp/files/461384/b6a6a05f-c50f-4fba-9fb3-e3a42bce4afe.dmp.txt)
http://www.xup.in/dl,11280759/b6a6a05f-c50f-4fba-9fb3-e3a42bce4afe.dmp/ (Upload here was not possible)

Regards

**Comment by mickem** [comment]
Not helpfull though...

I attached the crashing stacktrace below.
But as you can see the crash originates from some dot-net application...

Do you have any dot-net modules or similar loaded?
Could you provide the config?

```
Thread 23 (crashed)
 0  KERNELBASE.dll + 0x2a1c8
    rax = 0x000000d58a555de0   rdx = 0x0000000000000000
    rcx = 0x00007ffc4af36d4c   rbx = 0x000000d58a556df0
    rsi = 0x000000d58a557420   rdi = 0x000000d58a555c80
    rbp = 0x000000d58a555520   rsp = 0x000000d58a554e70
     r8 = 0x000000d58a558ff0    r9 = 0x00007ffc4a7e1a09
    r10 = 0x00007ffc4a7d0000   r11 = 0x00007ffc4b0d7914
    r12 = 0x000000d58a554fc8   r13 = 0x0000000000000000
    r14 = 0x000000d58a556298   r15 = 0x0000000000000000
    rip = 0x00007ffc80afa1c8
    Found by: given as instruction pointer in context
 1  KERNELBASE.dll + 0x2a1c8
    rbp = 0x000000d58a555520   rsp = 0x000000d58a554ea8
    rip = 0x00007ffc80afa1c8
    Found by: stack scanning
 2  0x7ffc4a7d0000
    rbp = 0x000000d58a555520   rsp = 0x000000d58a554ed8
    rip = 0x00007ffc4a7d0000
    Found by: stack scanning
 3  MSVCR120_CLR0400.dll + 0x25908
    rbp = 0x000000d58a555520   rsp = 0x000000d58a554ef0
    rip = 0x00007ffc75905908
    Found by: stack scanning
 4  MSVCR120_CLR0400.dll + 0x2587b
    rbp = 0x000000d58a555520   rsp = 0x000000d58a554f20
    rip = 0x00007ffc7590587b
    Found by: stack scanning
 5  MSVCR120_CLR0400.dll + 0x63095
    rbp = 0x000000d58a555520   rsp = 0x000000d58a554f40
    rip = 0x00007ffc75943095
    Found by: stack scanning
 6  MSVCR120_CLR0400.dll + 0x61e89
    rbp = 0x000000d58a555520   rsp = 0x000000d58a554f50
    rip = 0x00007ffc75941e89
    Found by: stack scanning
 7  clr.dll + 0x3d2790
    rbp = 0x000000d58a555520   rsp = 0x000000d58a554f58
    rip = 0x00007ffc4aba2790
    Found by: stack scanning
 8  clr.dll + 0x84861c
    rbp = 0x000000d58a555520   rsp = 0x000000d58a554fa0
    rip = 0x00007ffc4b01861c
    Found by: stack scanning
 9  clr.dll + 0x3d2790
    rbp = 0x000000d58a555520   rsp = 0x000000d58a554fb0
    rip = 0x00007ffc4aba2790
    Found by: stack scanning
10  clr.dll + 0x3d26db
    rbp = 0x000000d58a555520   rsp = 0x000000d58a555000
    rip = 0x00007ffc4aba26db
    Found by: stack scanning
11  ntdll.dll + 0x95c53
    rbp = 0x000000d58a555520   rsp = 0x000000d58a555020
    rip = 0x00007ffc83675c53
    Found by: stack scanning
12  MSVCR120_CLR0400.dll + 0x73820
    rbp = 0x000000d58a555520   rsp = 0x000000d58a555048
    rip = 0x00007ffc75953820
    Found by: stack scanning
13  MSVCR120_CLR0400.dll + 0xd19f0
    rbp = 0x000000d58a555520   rsp = 0x000000d58a555050
    rip = 0x00007ffc759b19f0
    Found by: stack scanning
14  mscorlib.ni.dll + 0x533781
    rbp = 0x000000d58a555520   rsp = 0x000000d58a555080
    rip = 0x00007ffc49833781
    Found by: stack scanning
15  clr.dll + 0x3d26db
    rbp = 0x000000d58a555520   rsp = 0x000000d58a555120
    rip = 0x00007ffc4aba26db
    Found by: stack scanning
```

**Comment by itsul** [comment]
Mainly default config:
# If you want to fill this file with all available options run the following command:
#   nscp settings --generate --add-defaults --load-all
# If you want to activate a module and bring in all its options use:
#   nscp settings --activate-module <MODULE NAME> --add-defaults
# For details run: nscp settings --help


; TODO
[/settings/default]

; Undocumented key
password = 123456

; Undocumented key
allowed hosts = 172.xx.xx.xx,127.0.0.1,::1

; TODO
[/settings/NRPE/server]

; Undocumented key
verify mode = none

; Undocumented key
insecure = true
allow arguments = true
payload length = 4096
allow nasty characters = true

; TODO
[/modules]

; Undocumented key
CheckExternalScripts = 1

; Undocumented key
CheckHelpers = 1

; Undocumented key
CheckEventLog = 1

; Undocumented key
CheckNSCP = 1

; Undocumented key
CheckDisk = 1

; Undocumented key
CheckSystem = 1

; Undocumented key
WEBSErver = 1

; Undocumented key
NRPEServer = 1

NRDPClient = 1

NSCAClient = 1

#[/settings/NSCA/client]
#channel = NSCA
#; The same host name configured in Icinga2
#hostname = auto

#[/settings/NSCA/client/targets/default]
#address = 172.27.11.17
#port = 5667
#allowed ciphers = ADH
#certificate = 
#encryption = 1
#password = 8ung4SuLis
#timeout = 30
#use ssl = false
#verify mode = none
[/settings/scheduler/schedules]

host_check=alias_check_ok

[/settings/external scripts/alias]

alias_check_ok = check_ok message="OK - HOST ALIVE"


[/settings/NRDP/client]
channel=NRDP
hostname=auto

[/settings/NRDP/client/targets/default]
address=xx.xx.xx.xx
host=xx.xx.xx.xx
port=80
retries=3
timeout=30
token=8ung4SuLis
template = Hello, this is %source% reporting %message%!


# EXTERNAL SCRIPT SECTION
# Section for external scripts configuration options (CheckExternalScripts).
[/settings/external scripts]
allow arguments = true
allow nasty characters = true
script path= sul\
timeout=60


[/settings/external scripts/scripts]
check_win_time=sul\check_windows_time.bat %ARGS%

**Comment by mickem** [comment]
One thing the WEBUI does is load all modules and I havehad other reports of this happening if the wrong dotnet framework version is installed...
Dopes it happen every time?
Which dot net version is inatlled?

And will deleting dotnetplugins.dll (in modules folder) resolve the issue...

**Comment by itsul** [comment]
My version is the following and yes it crashs every time.

![netframework](https://cloud.githubusercontent.com/assets/12328177/18552459/cf32e8fa-7b5c-11e6-8335-a46eafba3204.PNG)

**Comment by mickem** [comment]
I managed to reproduce the issue with 0.5.0.62 but it seems fixed as it (for me) works with the 0.5.1.19 build found here: https://github.com/mickem/nscp/releases/tag/0.5.1.19 Please reopen if you can reproduce this with the new build.
